iam_groups_authn/sync.py

```diff
@@ -51,21 +53,31 @@ def get_group_members(self, group):
             members: List of all members (groups or users) that belong to the IAM group.
         """
         # build service to call Admin SDK Directory API
-        service = build("admin", "directory_v1", credentials=self.creds)
+        if not self.creds.valid:
+            request = Request()
+            self.creds.refresh(request)
```

Comment:
nit: since you are doing this in a few areas, it's maybe worth adding a "authenticated_post" helper that takes a creds and a url and returns the result so you don't have this code in 3 different places